TCD1304.py

```diff
@@ -77,11 +73,9 @@ def analog_signal_read (self, block:False):
         self.scope.select_range('CH1', max_voltage_output)   # voltagerange should be fitted to the sensors output for better resolution. sensor otput is between 2V-3V due to datasheet
         self.scope.configure_trigger(channel = 'CH1', voltage = min_voltage_output ) # starts recording , when voltage is over a certain level
         self.scope.capture(channels=1, samples=integration_elements*samples_per_element, timegap=integration_time/samples_per_element, block=False)
-        self.icg_clock(self)
-        x, y= self.scope.fetch_data()
-        diff = abs(x,y[1, 0] - min_voltage_output )
-        analog_measurement = np.ndarray ()
-        return analog_measurement  # this should put the Measurements and timestamps in an array
+        icg_clock()
+        x, = self.scope.capture()  # putting timestamps into numpy array (don't know, if that is necessary for our purpose)
+        y, = self.scope.fetch_data()  # collecting the Voltage in nonblocking mode

```

Comment:
You probably want to return this.

Reply:
> # not needed: x, = self.scope.capture()

        y, = self.scope.fetch_data()  # collecting the Voltage in nonblocking mode
        return [y]; #returns samples

Correct?

Reply:
You could just return `y` as is, no need to put it in a list.

Reply:
fair point

Reply:
I corrected the "last" few things and committed it. 
Are there any more errors to fix?
How do we actually test the code? Freddi and myself are planning to write the report for the upcoming tuesday which means, it would be super cool to have some sort of a "goodie" for our professor.

Reply:
The first step to test the code is to check that you get some kind of output at all from the sensor. You have the sensor set up, right? What happens when you connect it to the PSLab and run this code?

If you get some output, the next step is to compare it to a known reference. A fluorescent light source with a known spectrum, for example. The output should match the reference.

As for the code, there are still a few errors (which I will address separately). Aside from the errors, there are some more things to do before the code can be merged:

- Usability: Consider how a user might be expected to use this module. What do they want to do? Is the code written in a way to make the user's task as simple as possible? For example, in order to get some output from the sensor, it is currently necessary to do something like this:
```python
from TCD1304 import TCD1304

spectrometer = TCD1304()
spectrometer.power_source()
spectrometer.master_clock()
spectrometer.sh_clock()
output = spectrometer.analog_signal_read()
```
Is there any reason why the user should have to manually power the device and set up the clocks? Why not start them automatically when the object is created? It would be more user friendly like this:
```python
from TCD1304 import TCD1304

spectrometer = TCD1304()
output = spectrometer.read()
```

- Various style issues:
  - Module- and class-level constants should be capitalized.
  - Function names should be verbs (e.g. `start_sh_clock` rather than `sh_clock`)
  - Function names should be as short and simple as possible while still being descriptive. In the method `analog_signal_read` there is no need to specify "analog" since the sensor cannot do a digital read. The name could be just `read` or `measure`.
- Documentation:
  - The module, class, and methods need docstrings explaining what they do. You currently have a lot of inline comments that could be moved to docstrings. Docstrings should be written in numpydoc style.